File: Memories.Server/Model/Role.cs
using System;
using System.Collections.Generic;

namespace Memories.Server.Model;

public partial class Role
{
    public int Code { get; set; }

    public string Name { get; set; } = null!;

    public virtual ICollection<User> IdUsers { get; set; } = new List<User>();
}

File: memories.client/src/model/role.ts
export default class Role {
  code: number | null;
  name: string;

  constructor(code: number | null, name: string) {
    this.code = code;
    this.name = name;
  }

  static generate(role: Role) {
    return new Role(role.code, role.name);
  }
}